api_types.ts

```diff
@@ -1295,7 +1328,9 @@ export interface SyncCompletionMetadata<IncrementalContinuationT = Continuation>
    * TODO(sam): Unhide this
    * @hidden
    */
-  hasIncompleteResults?: boolean;
+  hasIncompleteResults: true;
+
+  incrementalContinuation?: never;
```

Comment:
This doesn't seem right to me - our original intention with `hasIncompleteResults` was to give packs the option to continue incrementally syncing in a degraded state while a new full sync starts. cc @jonathan-codaio @sam-codaio

Reply:
Yeah this is wrong. We should be able to keep incrementally running while we have incomplete results of some kind. Otherwise we have to wait for a full sync for any updates which could take a days. We should keep syncing what we can since often this is just for edge cases where the API does not give us enough information for full fidelity

Reply:
I didn't see any packs returning both incremental results and `hasIncompleteResults: true`, which is why I made this change and why it didn't break anything. Most packs seem to abort at this point since they'd assume a full sync is triggered.

Reply:
Right, yeah we should allow for the case Sam/Patrick described, thanks for catching. Curious that it's not used in practice yet though.

Reply:
For the (apparently very common) case of a fatal error completely breaking an incremental sync, my thought was to model that as a new IncrementalSyncFatalError for packs to throw.